### PR TITLE
feat: vertex and image compression

### PR DESCRIPTION
**Pull request by cabljac**
This PR adds vertex to the two gemini extensions, and adds image compression for gemini-pro-vision when images are two large

### CODE REVIEW

File: firestore-genai-chatbot/README.md [hunk 1]
```diff
@@ -73,7 +73,7 @@ Additionally, this extension uses the Google AI Gemini API. For more details on
 
 **Configuration Parameters:**
 
-* Gemini API Provider: This extension makes use of the Gemini family of large language models. Currently the extension only supports the Google AI API (for developers) but in future will support the Vertex AI Gemini API.
+* Gemini API Provider: This extension makes use of the Gemini family of large language models. For Google AI you will require an API key, whereas Vertex AI will authenticate using application default credentials.
```
Comment (pr-Mais) [inline firestore-genai-chatbot/README.md:76]: Could add a resource for those not familiar with ADC, i.e. https://firebase.google.com/docs/admin/setup#initialize-sdk

File: firestore-genai-chatbot/extension.yaml [hunk 2]
```diff
@@ -75,12 +75,15 @@ params:
     label: Gemini API Provider
     description: >-
       This extension makes use of the Gemini family of large language models.
-      Currently the extension only supports the Google AI API (for developers)
-      but in future will support the Vertex AI Gemini API.
+      For Google AI you will require an API key, whereas Vertex AI will
+      authenticate using application default credentials. For more information
+      see the [docs]https://firebase.google.com/docs/admin/setup#initialize-sdk.
```
Comment (pr-Mais) [inline firestore-genai-chatbot/extension.yaml:80]:
```suggestion
      see the [docs](https://firebase.google.com/docs/admin/setup#initialize-sdk).
```

File: firestore-multimodal-genai/PREINSTALL.md [hunk 3]
```diff
@@ -53,9 +53,15 @@ This extension supports the following language models:
...
+This extension supports providing multimodal prompts. To use this feature, select the Gemini Pro Vision model on installation, and provide an Image Field parameter. The Image Field parameter should be the name of a document field in firestore.
 
-When you select these options, any document handled by the extension must contain an image field. The image field must be a string, and can either be the Cloud Storage URL of an object (e.g `gs://my-bucket.appspot.com/filename.png`) or the base64-encoded string of an image. This image will then be provided as part of the prompt to Gemini Pro Vision.
+When you select these options, any document handled by the extension must contain an image field. The image field must be a string, and can either be the Cloud Storage URL of an object (e.g `gs://my-bucket.appspot.com/filename.png`). This image will then be provided as part of the prompt to Gemini Pro Vision.
+
+The Gemini Pro Vision API has a limit on image sizes. For Google AI this limit is currently 1MB, and for Vertex AI this limit is 4MB. This extension will compress and resize images that fall above this limit.
+
+### Troubleshooting timeout/PROCESSING errors
```
Comment (pr-Mais) [inline firestore-multimodal-genai/PREINSTALL.md:62]: shouldn't this be in POSTINSTALL?

File: firestore-multimodal-genai/extension.yaml [hunk 4]
```diff
@@ -79,12 +81,15 @@ params:
     label: Gemini API Provider
     description: >-
       This extension makes use of the Gemini family of large language models.
-      Currently the extension only supports the Google AI API (for developers)
-      but in future will support the Vertex AI Gemini API.
+      For Google AI you will require an API key, whereas Vertex AI will
+      authenticate using application default credentials. For more information
+      see the [docs]https://firebase.google.com/docs/admin/setup#initialize-sdk.
```
Comment (pr-Mais) [inline firestore-multimodal-genai/extension.yaml:86]:
```suggestion
      see the [docs](https://firebase.google.com/docs/admin/setup#initialize-sdk).
```

**Review by pr-Mais — CHANGES_REQUESTED** [review]

File: firestore-multimodal-genai/PREINSTALL.md (hunk 3 above)
Comment (cabljac) [reply firestore-multimodal-genai/PREINSTALL.md:62]: hm yeah i agree

**Pull request merged**